### DISCUSSION TITLE
How to use findRaw or aggregateRaw with objectId?

### DISCUSSION THREAD

**Discussion by the-evgenii** [prisma/prisma#14683] [Q&A]
aggregateRaw/findRaw accepts InputJsonValue in turn is

`export type InputJsonValue = string | number | boolean | InputJsonObject | InputJsonArray`

 I want to aggregate by objectId

example
```JavaScript
import { ObjectId } from 'bson'

prisma.geo.aggregateRaw({
  pipeline: [
    { $match: { userId: new ObjectId(user.id) } }
  ]
})
```

how I can?

**Comment by austincrim** [comment]
Hey @the-evgeniyz 👋,

Have you tried passing the ObjectId as a plain string? What behavior do you get?

**Reply by the-evgenii** [reply]
Yes, shure. But i get an empty result. I can make request like this 
```JavaScript
{ $match: { $expr : { $eq: [ '$_id' , { $toObjectId: "61fc458b46d7874a3a97ef79" } ] } } }
```
but I think it is wrong way

**Reply by austincrim** [reply]
I tested this out locally and had no issues running this query:
```ts
let result = await prisma.geo.aggregateRaw({
  pipeline: [{ $match: { userId: '62faab2d75dcfbccf98368b5' } }]
})
```

Does that method work for you? If not, could you provide your full `schema.prisma` so I can recreate your setup?

**Reply by the-evgenii** [reply]
Hello, thank you for your answer
I think you are store userId as string not as `@db.ObjectId`

```Prisma
model User {
  id            String         @id @default(auto()) @map("_id") @db.ObjectId
  Geo           Geo[]
}

model Geo {
  id          String   @id @default(auto()) @map("_id") @db.ObjectId
  user        User     @relation(fields: [userId], references: [id])
  userId      String   @db.ObjectId
  createdAt   DateTime @db.Date
}
```

**Reply by pantharshit00** [reply]
Hello @the-evgenii 

I also wasn't able to reproduce this. Query worked fine for me as well:
<img width="1121" alt="image" src="https://user-images.githubusercontent.com/22195362/203323437-a276fa13-8f7e-4b46-aa59-ac38b34dcdae.png">

```ts
import { PrismaClient } from "@prisma/client";

async function main() {
  const prisma = new PrismaClient();

  let result = await prisma.geo.aggregateRaw({
    pipeline: [{ $match: { userId: "62faab2d75dcfbccf98368b5" } }],
  });
  console.log(result);
  prisma.$disconnect();
}

main();
```


Maybe try once with latest version and if this is still a problem, make a minimal script reproduction in a Git repo and share with us.

**Reply by pantharshit00** [reply]
Hello @the-evgenii,

Did you get a chance to try this out with the latest release? Please let me know if this issue is still relevant.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
